module/executiondatasync/optimistic_sync/execution_result_provider.go

```diff
@@ -12,6 +17,23 @@ type Criteria struct {
 	RequiredExecutors flow.IdentifierList
 }
 
+func NewCriteria(query *entities.ExecutionStateQuery) Criteria {
+	if query == nil {
+		// TODO: or return empty criteria?
+		return DefaultCriteria
```

Comment:
just return an empty criteria. this plays nice with the criteria resolution logic in `ExecutionResultProvider`

```suggestion
		return Criteria{}
```